applications/GeoMechanicsApplication/custom_strategies/strategies/residualbased_newton_raphson_strategy_linear_elastic_dynamic.hpp

```diff
@@ -0,0 +1,810 @@
+            non_converged_solutions.push_back(first);
+        }
+
+        if (is_converged) {
+            if (BaseType::mpConvergenceCriteria->GetActualizeRHSflag()) {
+                TSparseSpace::SetToZero(rb);
+
+                p_builder_and_solver->BuildRHS(p_scheme, r_model_part, rb);
+            }
+
+            is_converged =
+                BaseType::mpConvergenceCriteria->PostCriteria(r_model_part, r_dof_set, rA, rDx, rb);
+        }
+
+        // Iteration Cycle... performed only for non linear RHS
+        if (!is_converged) {
+            is_converged = this->PerformIterationCycle(rA, rDx, rb, dx_tot, non_converged_solutions, iteration_number);
+        }
+
+        if (is_converged) {
+            this->UpdateSolutionStepDerivative(dx_tot, r_model_part);
+        }
+
+        // plots a warning if the maximum number of iterations is exceeded
+        if (iteration_number >= BaseType::mMaxIterationNumber) {
+            this->MaxIterationsExceeded();
+        } else {
+            KRATOS_INFO_IF("ResidualBasedNewtonRaphsonStrategyLinearElasticDynamic", this->GetEchoLevel() > 0)
+                << "Convergence achieved after " << iteration_number << " / "
+                << BaseType::mMaxIterationNumber << " iterations" << std::endl;
+        }
+
+        // calculate reactions if required
+        if (BaseType::mCalculateReactionsFlag)
+            p_builder_and_solver->CalculateReactions(p_scheme, r_model_part, rA, dx_tot, rb);
+
+        if (BaseType::mStoreNonconvergedSolutionsFlag) {
+            BaseType::mNonconvergedSolutionsMatrix =
+                Matrix(r_dof_set.size(), non_converged_solutions.size());
+            for (std::size_t i = 0; i < non_converged_solutions.size(); ++i) {
+                block_for_each(r_dof_set, [non_converged_solutions, i, this](const auto& r_dof) {
+                    BaseType::mNonconvergedSolutionsMatrix(r_dof.EquationId(), i) =
+                        non_converged_solutions[i](r_dof.EquationId());
+                });
+            }
+        }
+
+        return is_converged;
+    }
+
+    /**
+     * @brief This method provides the defaults parameters to avoid conflicts between the different constructors
+     * @return The default parameters
+     */
+    Parameters GetDefaultParameters() const override
+    {
+        auto default_parameters = Parameters(R"(
+        {
+            "name"                                : "newton_raphson_strategy_linear_elastic_dynamic",
+            "use_old_stiffness_in_first_iteration": false,
+            "max_iteration"                       : 10,
+            "reform_dofs_at_each_step"            : false,
+            "compute_reactions"                   : false,
+            "builder_and_solver_settings"         : {},
+            "convergence_criteria_settings"       : {},
+            "linear_solver_settings"              : {},
+            "scheme_settings"                     : {}
+        })");
```

Comment:
Here, only the name is different from the base. Since we apply the base defaults here anyway, wouldn't this be enough?
```suggestion
        auto default_parameters = Parameters(R"(
        {
            "name"                                : "newton_raphson_strategy_linear_elastic_dynamic"
        })");
```